src/compiler/crystal/macros/methods.cr

```diff
@@ -2415,6 +2415,54 @@ module Crystal
     end
   end
 
+  class ClassDef
+    def interpret(method : String, args : Array(ASTNode), named_args : Hash(String, ASTNode)?, block : Crystal::Block?, interpreter : Crystal::MacroInterpreter, name_loc : Location?)
+      case method
+      when "kind"
+        interpret_check_args { MacroId.new(@struct ? "struct" : "class") }
+      when "name"
+        interpret_check_args(named_params: ["generic_args"]) do
+          if parse_generic_args_argument(self, method, named_args, default: true) && (type_vars = @type_vars)
+            type_vars = type_vars.map_with_index do |type_var, i|
+              param = MacroId.new(type_var)
+              param = Splat.new(param) if i == @splat_index
+              param
+            end
+            Generic.new(@name, type_vars)
+          else
+            @name
+          end
```

Comment:
**polish:** This feels complex enough that it could be extracted to a helper method. It also seems very similar to the implementation for `ModuleDef`, so extracting it makes it reusable.